test/validation/api/ipsec/ipsec.c

```diff
@@ -1266,6 +1266,46 @@ void ipsec_test_packet_from_pkt(ipsec_test_packet *test_pkt, odp_packet_t *pkt)
 	odp_packet_free(*pkt);
 }
 
+void rebuild_ethernet_header(ipsec_test_packet *pkt)
+{
+	odph_ethhdr_t eth = {
+		.dst.addr = {0x02, 0x00, 0x00, 0x00, 0x00, 0x01},
+		.src.addr = {0x02, 0x00, 0x00, 0x00, 0x00, 0x02},
+	};
+	uint32_t l3_len;
+
+	CU_ASSERT_FATAL(pkt->l3_offset != ODP_PACKET_OFFSET_INVALID);
+	CU_ASSERT_FATAL(pkt->len > pkt->l3_offset);
+	l3_len = pkt->len - pkt->l3_offset;
+	CU_ASSERT_FATAL(ODPH_ETHHDR_LEN + l3_len < sizeof(pkt->data));
+
+	if (pkt->l3_offset != ODPH_ETHHDR_LEN)
+		memmove(&pkt->data[ODPH_ETHHDR_LEN], &pkt->data[pkt->l3_offset], l3_len);
+
+	switch (ODPH_IPV4HDR_VER(pkt->data[ODPH_ETHHDR_LEN])) {
+	case 4:
```

Comment:
We seem to have `ODPH_IPV4` and `ODPH_IPV6`.

Reply:
used those in v2